src/Compression/CompressionCodecDeflate.h

```diff
@@ -0,0 +1,245 @@
+        {
+            return -1;
+        }
+        auto status = qpl_init_job(PATH, qpl_job_ptr);
+        if (status != QPL_STS_OK)
+        {
+            return -1;
+        }
+        return 0;
+    }
+
+    int32_t ALWAYS_INLINE initJobPool()
+    {
+        static bool initialized = false;
+
+        if (initialized == false)
+        {
+            const int32_t size = get_job_size_helper();
+            if (size < 0)
+                return -1;
+            for (int i = 0; i < jobPoolSize; ++i)
+            {
+                jobPool[i] = nullptr;
+                qpl_job * qpl_job_ptr = reinterpret_cast<qpl_job *>(new uint8_t[size]);
+                if (init_job_helper(qpl_job_ptr) < 0)
+                    return -1;
+                jobPool[i] = qpl_job_ptr;
+                jobLock[i].store(false);
+            }
+            initialized = true;
+        }
+        return 0;
+    }
+
+    bool ALWAYS_INLINE tryLockJob(size_t index)
+    {
+        bool expected = false;
+        return jobLock[index].compare_exchange_strong(expected, true);
+    }
+
+    void ALWAYS_INLINE destroyJobPool()
+    {
+        const uint32_t size = get_job_size_helper();
+        for (uint32_t i = 0; i < jobPoolSize && size > 0; ++i)
+        {
+            while (tryLockJob(i) == false)
+            {
+            }
+            if (jobPool[i])
+            {
+                qpl_fini_job(jobPool[i]);
+                delete[] jobPool[i];
+            }
+            jobPool[i] = nullptr;
+            jobLock[i].store(false);
+        }
+    }
+
+    struct ReleaseJobObjectGuard
+    {
+        uint32_t index;
+        ReleaseJobObjectGuard() = delete;
+
+    public:
+        ALWAYS_INLINE ReleaseJobObjectGuard(const uint32_t i) : index(i)
+        {
+        }
+        ALWAYS_INLINE ~ReleaseJobObjectGuard()
+        {
+            jobLock[index].store(false);
+        }
+    };
+    Poco::Logger * log;
+};
+class SoftwareCodecDeflate
+{
+public:
+    SoftwareCodecDeflate();
+    ~SoftwareCodecDeflate();
+    uint32_t doCompressData(const char * source, uint32_t source_size, char * dest, uint32_t dest_size);
+    void doDecompressData(const char * source, uint32_t source_size, char * dest, uint32_t uncompressed_size);
+
+private:
+    qpl_job * jobSWPtr; //Software Job Codec Ptr
+    std::unique_ptr<uint8_t[]> jobSWbuffer;
+    qpl_job * getJobCodecPtr();
+};
+
+class HardwareCodecDeflate
+{
+public:
+    bool hwEnabled;
+    HardwareCodecDeflate();
+    ~HardwareCodecDeflate();
+    uint32_t doCompressData(const char * source, uint32_t source_size, char * dest, uint32_t dest_size) const;
+    uint32_t doDecompressData(const char * source, uint32_t source_size, char * dest, uint32_t uncompressed_size) const;
+    uint32_t doDecompressDataReq(const char * source, uint32_t source_size, char * dest, uint32_t uncompressed_size);
+    void doDecompressDataFlush();
+
+private:
+    std::map<uint32_t, qpl_job *> jobDecompAsyncMap;
```

Comment:
Not immediately clear what the uint32 keys stand for, pls document.

Reply:
Added comment into the code as below: 
   /// Asynchronous job map for decompression: job ID - job object.
    /// For each submission, push job ID && job object into this map;
    /// For flush, pop out job ID && job object from this map. Use job ID to release job lock and use job object to check job status till complete.